c/misra/test/rules/RULE-8-4/object1.c

```diff
@@ -5,6 +5,7 @@ extern int i1 = 0; // NON_COMPLIANT
 
 int i2 = 0; // NON_COMPLIANT
 
-extern int i3; // NON_COMPLIANT
+// not detected as `short i3` exists
+extern int i3; // NON_COMPLIANT[FALSE_NEGATIVE]
```

Comment:
I did not have this particular reviewed on the `next` branch, as I'm not sure that just updating the tests here is the right solution (and needed to get `next` working again).  It might be that the proper solution is to improve the query to also make sure that the types of all the `DeclarationEntry`s are identical.

Reply:
Great!

Some context here from the CodeQL side. We between 2.9.4 and 2.10.5 we added some magic that unifies declaration entries of variables and presents them as being part of a single declaration when they all have the same name and we see that there's only one definition of the variable in the whole database. This allows us to automatically do some limited reasoning across link targets.

Reply:
Indeed that one.